utils/downloads.py

```diff
@@ -53,3 +54,30 @@ def download_sounds(licenses_url, pack):
     response = HttpResponse(filelist, content_type="text/plain")
     response['X-Archive-Files'] = 'zip'
     return response
+
+
+def should_suggest_donation(user, times_displayed):
+    """
+    This method indicates when we should display the donation popup to the
+    user. This will be based on 3 settings indicating how many days
+    after a donation we show the popup again, after how many downloads we
+    display the popup and for how long. The popup will be shown a number of
+    times per day.
+    """
+
+    if times_displayed and times_displayed > settings.DONATION_MODAL_DISPLAY_TIMES_DAY:
+        return False
+
+    last_donation = user.donation_set.order_by('created').last()
```

Comment:
I always find it annoying if variables are created 6 lines before they are used. This line immediately made me think: hey, what if the user never donated. Suggestion: move this to before the if.